lib/src/Uncertainty/Algorithm/MetaModel/MetaModelValidation.cxx

```diff
@@ -179,24 +206,39 @@ GridLayout MetaModelValidation::drawValidation() const
 void MetaModelValidation::save(Advocate & adv) const
 {
   PersistentObject::save(adv);
-  adv.saveAttribute( "inputSample_", inputSample_ );
   adv.saveAttribute( "outputSample_", outputSample_ );
-  adv.saveAttribute( "metaModel_", metaModel_ );
-  adv.saveAttribute( "isInitialized_", isInitialized_ );
+  adv.saveAttribute( "metaModelPredictions_", metaModelPredictions_ );
   adv.saveAttribute( "residual_", residual_ );
-  adv.saveAttribute( "q2_", q2_ );
+  adv.saveAttribute( "r2Score_", r2Score_ );
+  adv.saveAttribute( "meanSquaredError_", meanSquaredError_ );
 }
 
 /* Method load() reloads the object from the StorageManager */
 void MetaModelValidation::load(Advocate & adv)
 {
   PersistentObject::load(adv);
-  adv.loadAttribute( "inputSample_", inputSample_ );
   adv.loadAttribute( "outputSample_", outputSample_ );
-  adv.loadAttribute( "metaModel_", metaModel_ );
-  adv.loadAttribute( "isInitialized_", isInitialized_ );
+  if (adv.hasAttribute("metaModelPredictions_"))
+    adv.loadAttribute( "metaModelPredictions_", metaModelPredictions_ );
+  else
+    adv.loadAttribute( "outputSample_", metaModelPredictions_ );
   adv.loadAttribute( "residual_", residual_ );
-  adv.loadAttribute( "q2_", q2_ );
+  if (adv.hasAttribute("r2Score_"))
+    adv.loadAttribute( "r2Score_", r2Score_ );
+  else
+    adv.loadAttribute( "q2_", r2Score_ );
+  if (adv.hasAttribute("meanSquaredError_"))
+    adv.loadAttribute( "meanSquaredError_", meanSquaredError_ );
+  else
+  {
+    const UnsignedInteger outputDimension = outputSample_.getDimension();
+    if (r2Score_.getDimension() != outputDimension)
+      throw NotDefinedException(HERE) << "The dimension of the R2 score is " << r2Score_.getDimension() 
+                                      << " but the dimension of the output sample is " << outputDimension;
+    const Point sampleVariance(outputSample_.computeCentralMoment(2));
+    for (UnsignedInteger j = 0; j < outputDimension; ++ j)
+      meanSquaredError_[j] = sampleVariance[j] * (1.0 - r2Score_[j]);
+  }
```

Comment:
If you try to load an old version of the object where q2_ has not been computed before to be saved, the object you load is not correct as you don't compute r2Score_. You should write:
```
if (adv.hasAttribute("r2Score_"))
    adv.loadAttribute( "r2Score_", r2Score_ );
 if (adv.hasAttribute("meanSquaredError_"))
    adv.loadAttribute( "meanSquaredError_", meanSquaredError_ );
else // Old version of the object
{
  // Recompute everything as meanSquaredError_ has to be recomputed in any case
  Sample inputSample;
  Function metaModel;
  adv.loadAttribute( "inputSample_", inputSample );
  adv.loadAttribute( "metaModel_", metaModel );
  *this = MetaModelValidation(outputSample_, metaModel(inputSample));
} // Old version of the object
```
IMO it is not worse the burden to check if residual_ and q2_ were correct (using isInitialized_) in order to reduce the computation to meanSquaredError_, it is better to ensure that everything is conform to the new implementation if it has been detected that the object was saved in an old version.

Reply:
Ok. Fixed at c38c9197a283eb00855942634c7eefbc006a405b.

It used a mix of @jschueller and @regislebrun suggestions.
It checked with a Study created unitialized with OT 1.22 and loaded with master: it works.